crates/chain/src/keychain/txout_index.rs

```diff
@@ -588,3 +588,63 @@ where
         .take_while(Result::is_ok)
         .map(Result::unwrap)
 }
+
+pub struct SpkIterator<'a> {
+    next_index: u32,
+    end: u32,
```

Comment:
Since `end` is always 0 or `BIP32_MAX_INDEX` I don't think you need the `end` field. You only need to check in the logic (as you're doing) if `descriptor.has_wildcard()`. I have a suggestion below moving the logic to the `nth()` function.

Reply:
@notmandatory I don't think this is correct, I see where you are coming from, but sometimes we want to bound the iterator to a specific range. Having the `end` index here would be helpful for that, unless if you have a better idea in mind?

Reply:
Agreed!

Reply:
We have a second constructor `new_with_range` for this now!